src/screens/DiscoveryScreenV2/index.js

```diff
@@ -51,13 +51,21 @@ const DiscoveryScreenV2 = ({route}) => {
   const [profileState] = React.useContext(Context).profile;
   const cancelTokenRef = React.useRef(axios.CancelToken.source());
   const [, followingDispatch] = React.useContext(Context).following;
+  const [userPage, setUserPage] = React.useState({
+    currentPage: 1,
+    limitPage: 1
+  });
+  const [topicPage, setTopicPage] = React.useState({
+    currentPage: 1,
+    limitPage: 1
+  });
+  const [domainPage, setDomainPage] = React.useState({
+    currentPage: 1,
+    limitPage: 1
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The state variables `userPage`, `topicPage`, and `domainPage` are initialized with the same structure. This is a repetition of code and can be abstracted into a function to improve maintainability.
```diff
-  const [userPage, setUserPage] = React.useState({
-    currentPage: 1,
-    limitPage: 1
-  });
-  const [topicPage, setTopicPage] = React.useState({
-    currentPage: 1,
-    limitPage: 1
-  });
-  const [domainPage, setDomainPage] = React.useState({
-    currentPage: 1,
-    limitPage: 1
-  });

+  const initializePageState = () => ({
+    currentPage: 1,
+    limitPage: 1
+  });
+  
+  const [userPage, setUserPage] = React.useState(initializePageState());
+  const [topicPage, setTopicPage] = React.useState(initializePageState());
+  const [domainPage, setDomainPage] = React.useState(initializePageState());
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->